tests/neg-macros/quote-this.scala

```diff
@@ -12,12 +12,14 @@ class Foo {
   }
 
   inline def i(): Unit = ${ Foo.impl[Any]('{
-    given QuoteContext = ???
+    val x: QuoteContext = ???
+    given x.type = x
```

Comment:
/cc @odersky : this seems to be a friction between `given` and path-dependent types.